session/pessimistic_test.go

```diff
@@ -703,10 +709,15 @@ func (s *testPessimisticSuite) TestInnodbLockWaitTimeout(c *C) {
 		tk5.MustExec("set innodb_lock_wait_timeout = 2")
 		tk5.MustExec("begin pessimistic")
 		_, err := tk5.Exec("update tk set c2 = c2 - 1 where c1 = 1")
-		c.Check(err.Error(), Equals, tikv.ErrLockWaitTimeout.Error())
+		timeoutErrCh <- err
 		tk5.MustExec("rollback")
 	}()
 
+	timeoutErr := <-timeoutErrCh
+	c.Assert(timeoutErr.Error(), Equals, tikv.ErrLockWaitTimeout.Error())
```

Comment:
I mean, make sure err is not null before calling err.Error(), otherwise may lead to panic.